framework/source/class/qx/util/DynamicScriptLoader.js

```diff
@@ -0,0 +1,232 @@
+     EVENTS
+  *****************************************************************************
+  */
+
+  events: {
+    /**
+     * fired when a script is loaded successfully. The data contains 'script' and 'status' keys.
+     */
+    loaded: 'qx.event.type.Data',
+
+    /**
+     * fired when a specific script fails loading.  The data contains 'script' and 'status' keys.
+     */
+    failed: 'qx.event.type.Data',
+
+    /**
+     * fired when all given scripts are loaded, each time loadScriptsDynamic is called.
+     */
+    ready: 'qx.event.type.Event'
+  },
+
+  statics: {
+    __ADDED: {},
+    __LOADED: {}
+  },
+
+  /*
+  *****************************************************************************
+     MEMBERS
+  *****************************************************************************
+  */
+
+
+  members: {
+
+    __QUEUE: null,
+    __LOADING: null,
+
+
+    /**
+     * Load scripts dynamically, typically used in the instance constructor
+     *
+     * @param codeArr {Array} an array with the uri names of the scripts
+     * @param callback {Function?undefined} a callback function to call when all the scripts in the array have been loaded
+     * @param context {Object?undefined} a contex for calling the callback
+     *
+     * @return {String|null} name of the last script to be loaded
+     *     Returns the name of the last script you can expect to be loaded if all scripts are already loaded null will be returned.
+     *
+     */
+    load: function(codeArr) {
+      codeArr.forEach(function(script) {
+         this.__QUEUE.push(script);
+      },this);
+      this.__loadScripts();
+    },
+
+
+    /**
+     * Chain loading scripts.
+     *
+     * Recursively called until the array of scripts is consumed
+     *
+     * @param codeArr {Array} an array with the uri names of the scripts
+     */
+    __loadScripts: function() {
+      var cl = qx.util.DynamicScriptLoader;
+      if (this.__LOADING === true) {
+        return;
+      }
+
+      var script = this.__QUEUE.shift();
+      if (!script){
+        this.fireEvent("ready")
+        return;
+      }
+
+      if (cl.__LOADED[script]){
+        this.fireDataEvent('loaded',{
+          script: script,
+          status: 'preloaded'
+        });
+        this.__loadScripts();
+        return;
+      }           
+
+      this.__LOADING = true
+
+      var dynLoader = cl.__ADDED[script];
+      if (dynLoader){
+          var id1 = dynLoader.addListener('loaded',function(e){
+            var data = e.getData();
+            if (data.script === script){
+              dynLoader.removeListenerById(id2);
+              dynLoader.removeListenerById(id1);
+              this.fireDataEvent('loaded',data);
+              this.__LOADING = false;
+              this.__loadScripts();
+            }
+          },this);
+          var id2 = dynLoader.addListener('failed',function(e){
+            var data = e.getData();
```

Comment:
There is, unfortunately, no standard for that in qooxdoo. Personally, I find it much easier to know what's going on in my function and especially what scope a variable is declared in, if I need only look for it at the top of each function.